keras/src/layers/activations/softmax.py

```diff
@@ -50,6 +50,12 @@ def __init__(self, axis=-1, **kwargs):
 
     def call(self, inputs, mask=None):
         if mask is not None:
+            if mask.shape != inputs.shape:
+                raise ValueError(
+                    "`mask` and `inputs` must have same shape. "
```

Comment:
Actually documentation states that mask should be of same shape. May be we need to add here that same shape or broadcastable shape.
```
mask: A boolean mask of the same shape as `inputs`. The mask
          specifies 1 to keep and 0 to mask. Defaults to `None`.
```

Also the first dimension of shape i.e `batch_size` should not be considered in shape. Because if we pass `input_shape` as (5, 4, 1) and mask as (5,4) we get error as "`Incompatible shapes: [5,4,1] vs. [5,4]`". Do we need to mention this explicitly in docs?

Also the compute_output_shape method simply returns input_shape. Suppose if I pass 

```
x = np.random.rand(7, 1, 5)
mask = np.ones((5, 1))
l_1 = keras.layers.Softmax(axis=-1,trainable = True,dtype='float32',autocast = True)
print(l_1(x).shape) #(7, 1, 5)
print(l_1(x,mask).shape) #(7, 5,1) 
print(l_1.compute_output_shape(x.shape)) #(7, 1, 5)
```
Noticed output shape is different from `compute_output_shape` method as the `mask` is broadcastable here.
If I change mask shape to (1,5) instead of (5,1) then all shapes are same.

Reply:
Hi @fchollet Any update on this PR? Please. Thank you!